### PR TITLE
Remove the type parameter from AbstractPage

### PR DESCRIPTION
**Pull request by MidnightDesign**
Signed-off-by: Rudolph Gottesheim <r.gottesheim@midnight-design.at>

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes?
| BC Break      | no
| New Feature   | no
| RFC           | yes?
| QA            | yes?

### Description

Follow-up from the discussion in https://github.com/laminas/laminas-navigation/pull/40#issuecomment-1339442990

### CODE REVIEW

File: src/Navigation.php [hunk 1]
```diff
@@ -12,8 +12,7 @@
 /**
  * A simple container class for {@link Laminas\Navigation\Page} pages
  *
- * @template TPage of AbstractPage
- * @template-extends AbstractContainer<TPage>
+ * @template-extends AbstractContainer<AbstractPage>
  */
 class Navigation extends AbstractContainer
```
Comment (Ocramius) [inline src/Navigation.php]: If I get this correctly, with this change, `AbstractContainer` becomes extensible (accepting any kind of page), whereas `Navigation` only supports `AbstractPage`?

I suppose that's workable :thinking:

Comment (MidnightDesign) [reply src/Navigation.php]: Actually, now that you mention it, I don't think there's any value in `AbstractContainer` being generic either. But on the other hand, it isn't hurting anything. I don't know, what do you say?

Comment (internalsystemerror) [reply src/Navigation.php]: I'd say this is the right approach, it's only the genericness of the page as an issue as the generic type is used to specify child pages of a page, which could be anything that extends `AbstractPage`, it doesn't need to be restricted by a generic IMO.

Comment (internalsystemerror) [reply src/Navigation.php]: Although actually, I was thinking this change would only be made to `src/Page/*.php` and not the containers. I'd keep the generic on this `Navigation` class.

Comment (MidnightDesign) [reply src/Navigation.php]: Fair. I'll put the `@template` back into `Navigation`.

**Review by internalsystemerror — APPROVED** [review]
LGTM. @Ocramius good to ship?

**Comment by Ocramius** [comment]
Yes, take it away 👍

**Comment by MidnightDesign** [comment]
Thank you guys!

**Comment by internalsystemerror** [comment]
Thanks @MidnightDesign. Released as [2.18.1](https://github.com/laminas/laminas-navigation/releases/tag/2.18.1)

**Pull request merged**